modin/data_management/partitioning/partition_collections.py

```diff
@@ -682,6 +686,43 @@ class RayBlockPartitions(BlockPartitions):
     def __init__(self, partitions):
         self.partitions = partitions
 
+    # We override these for performance reasons.
```

Comment:
Is this the critical change that made it work?

Reply:
We actually cache the lengths themselves with this structure. We also cache the object id's from the lengths right now so we can do a `ray.get` on them also. So to answer your question, we already do cache the result of the remote function.